packages/logs/src/domain/assembly.ts

```diff
@@ -0,0 +1,89 @@
+import type { Context, RawError, RelativeTime } from '@datadog/browser-core'
+import { ErrorSource, combine, createEventRateLimiter, getRelativeTime } from '@datadog/browser-core'
+import type { CommonContext } from '../rawLogsEvent.types'
+import type { LogsConfiguration } from './configuration'
+import type { LifeCycle } from './lifeCycle'
+import { LifeCycleEventType } from './lifeCycle'
+import type { Logger } from './logger'
+import { HandlerType, StatusType } from './logger'
+import { isAuthorized } from './logsCollection/logger/loggerCollection'
+import type { LogsSessionManager } from './logsSessionManager'
+import { reportRawError } from './reportRawError'
+
+export function startLogsAssembly(
+  sessionManager: LogsSessionManager,
+  configuration: LogsConfiguration,
+  lifeCycle: LifeCycle,
+  getCommonContext: () => CommonContext,
+  mainLogger: Logger // Todo: [RUMF-1230] Remove this parameter in the next major release
+) {
+  const reportAgentError = (error: RawError) => reportRawError(error, lifeCycle)
+
+  const logRateLimiters = {
+    [StatusType.error]: createEventRateLimiter(
+      StatusType.error,
+      configuration.eventRateLimiterThreshold,
+      reportAgentError
+    ),
+    [StatusType.warn]: createEventRateLimiter(
+      StatusType.warn,
+      configuration.eventRateLimiterThreshold,
+      reportAgentError
+    ),
+    [StatusType.info]: createEventRateLimiter(
+      StatusType.info,
+      configuration.eventRateLimiterThreshold,
+      reportAgentError
+    ),
+    [StatusType.debug]: createEventRateLimiter(
+      StatusType.debug,
+      configuration.eventRateLimiterThreshold,
+      reportAgentError
+    ),
+    ['custom']: createEventRateLimiter('custom', configuration.eventRateLimiterThreshold, reportAgentError),
+  }
+
+  lifeCycle.subscribe(
+    LifeCycleEventType.RAW_LOG_COLLECTED,
+    ({ rawLog, messageContext = undefined, savedCommonContext = undefined, logger = mainLogger }) => {
+      const startTime = rawLog.date ? getRelativeTime(rawLog.date) : undefined
+      const session = sessionManager.findTrackedSession(startTime)
+
+      if (!session) {
+        return
+      }
+
+      const commonContext = savedCommonContext || getCommonContext()
+      const log = combine(
+        { service: configuration.service, session_id: session.id },
+        { date: commonContext.date, view: commonContext.view },
+        commonContext.context,
+        getRUMInternalContext(startTime),
+        rawLog,
+        logger.getContext(),
+        messageContext
+      )
```

Comment:
It would mean that at assembly level, we can not guarantee the format for `rawLogsEvent` coming from `Loggers` since it can be overridden by the user. This can be problematic for logic relying on certain field like `date`, `origin` or `status`. Wdyt?

Reply:
FMU, those fields can still be overwritten by the customer. I would find it less surprising to use customer value as soon as possible rather than using our value for an internal behavior (like associate RUM context from the date) and then overwrite our value by the customer one which would make thing quite hard to debug.

Reply:
Fair enough. Since it's a non negligible change I'm in favor of doing it when tackling the [RUMF-1230](https://datadoghq.atlassian.net/browse/RUMF-1230)
Wdyt?